app/models/course.rb

```diff
@@ -538,6 +539,14 @@ def validate_enrichment_publishable
     validate_enrichment :publish
   end
 
+  def validate_site_statuses_publishable
+    site_statuses.each do |site_status|
+      unless site_status.valid?
+        raise RuntimeError.new("Site status invalid: #{site_status.errors.full_messages.first}")
```

Comment:
Can we add info about which provider and location(s) are invalid? At the moment, pulling the URL out for a request isn't as straightforward as it should be, and that's the only place I can think off, offhand, where we'd have that if we get this error.